### PR TITLE
feat: Update example to match current EKS blueprints, remove experimental feature to allow users to use Terraform 1.3+

### PR DESCRIPTION
**Pull request by bryantbiggs**
- Remove `module_variable_optional_attrs` which is removed in v1.3.0 of Terraform; this currently is holding back EKS Blueprints from being able to suport - https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/988
	- Closes #7
- General updates to better align with current EKS Blueprints project - see notes below and throughout changeset

### CODE REVIEW

File: blueprints/getting-started/main.tf
Comment (bryantbiggs) [inline blueprints/getting-started/main.tf:41]: pin version for repeatability/stability

File: blueprints/getting-started/main.tf
Comment (bryantbiggs) [inline blueprints/getting-started/main.tf:70]: pin version for repeatability/stability

File: blueprints/getting-started/main.tf
Comment (bryantbiggs) [inline blueprints/getting-started/main.tf:91]: We shouldn't depend on other modules and instead map the dependency ordering by passing outputs to the modules inputs for implicit dependency mapping. 

Using `depends_on` here will cause disruptions for users, specifically around IAM policies which cause service disruption - ref https://github.com/hashicorp/terraform/issues/30340#issuecomment-1010202582

File: locals.tf
Comment (bryantbiggs) [inline locals.tf:3]: `file()` works here when we aren't templating and simply reading in a file

File: locals.tf
Comment (bryantbiggs) [inline locals.tf:22]: we can take advantage of things like `lookup()` or `try()` to first attempt to get the value from `var.helm_config` if the user has provided one, otherwise fall back to a default value. When falling back to a default value and we don't have an opinion on an optional argument, we should default to `null` to indicate as such and leave it up to the resource's defaults

File: main.tf
Comment (bryantbiggs) [inline main.tf:8]: this should be hardcoded, its not configurable nor optional

File: terraform.tf
Comment (bryantbiggs) [inline terraform.tf:5]: since we are using the helm provider for `helm_release`, we should declare that the helm provider is required - https://www.terraform.io/language/providers/requirements#requiring-providers

File: variables.tf
Comment (bryantbiggs) [inline variables.tf:1]: https://github.com/terraform-linters/tflint is great for finding un-used variables/locals and other common conventions

**Comment by bryantbiggs** [comment]
@ksatirli would you mind taking a look when you get a chance so we can resolve the upstream issue related to Terraform v1.3+

File: blueprints/getting-started/main.tf
Comment (ksatirli) [reply blueprints/getting-started/main.tf:2]: Great sleuthing and fix!

**Comment by ksatirli** [comment]
@bryantbiggs these look great on first look, I'll take an in-depth look in the morning. 

Excited about the changes you're proposing (with `vault_namespace` being the _only_ thing that immediately stood out as a case of [unfortunate naming](https://developer.hashicorp.com/vault/docs/enterprise/namespaces))

File: blueprints/getting-started/main.tf
Comment (bryantbiggs) [reply blueprints/getting-started/main.tf]: removing pin due to cyclical conflict

**Comment by bryantbiggs** [comment]
Thanks @ksatirli - I've updated to change `vault_namespace` to simply `namespace` in the example. I also removed the version pin due to a cyclical conflict between external modules and the Blueprints repository/version

File: locals.tf
Comment (ksatirli) [reply locals.tf:3]: It does - I think originally, we planned on templatizing this more, but then decided not to.

Thanks for cleaning this up!

**Comment by ksatirli** [comment]
> Thanks @ksatirli - I've updated to change `vault_namespace` to simply `namespace` in the example. I also removed the version pin due to a cyclical conflict between external modules and the Blueprints repository/version

Thanks - These changes look great!

I vaguely remember there being a reason to _not_ pin the Blueprints module (which is - as you point out - an anti-pattern), but I can't find the actual _why_.

**Comment by bryantbiggs** [comment]
Yes, I think the way that we have both the module definition for the `helm_addon` and the module implementation in the same repository coupled with the fact that we have external addons like Vault that also point to this addon and are referenced from within the same project, it creates a nasty cyclical loop. I will have a chat with the team to see if we can do something about moving the `helm_addon` to its own module repo to break this loop and then we can utilize versions better

File: variables.tf
Comment (ksatirli) [reply variables.tf:1]: I dug into this and am grateful for you spotting this and reporting it!

The Superlinter [config](https://github.com/hashicorp/terraform-aws-hashicorp-vault-eks-addon/blob/main/.github/workflows/superlinter.yml#L27) for this repository has `VALIDATE_TERRAFORM` enabled and that used to take care of `tflint`. 

As it turns out, in `v4` of Superlinter, `VALIDATE_TERRAFORM_TFLINT` is the correct way for `tflint` to be enabled, as more granular control was added.

I've fixed this in the long-running PR of https://github.com/hashicorp/terraform-aws-hashicorp-vault-eks-addon/pull/3 and also merged this after brief discussion.

For other readers: #3 has a couple of `tflint` warnings that are fixed by this PR (#8)

**Comment by ksatirli** [comment]
@bryantbiggs I've tested this locally with `1.3.1` and found it working. 

I've enabled `tflint` via #3, which caused a small merge conflict in `blueprints/getting-started/main.tf` - happy to resolve this or let you be the sole committer of this PR.

**Comment by ksatirli** [comment]
Code looks great @bryantbiggs - the `terraform-docs` Action will be fixed in an upcoming PR.

Unless you need any more changes, I'm good with merging this.

**Review by ksatirli — APPROVED** [review]

**Comment by ksatirli** [comment]
Time to :shipit:.

Thank you once again, @bryantbiggs

**Pull request merged**
